### ISSUE TITLE
Fix Widget

### ISSUE THREAD

**Issue by xmha97** [deckerst/aves#343]
Please fit the widget in the frame.

![Screenshot_2022-09-28-02-46-47-876_com.microsoft.launcher.jpg](https://user-images.githubusercontent.com/23485114/192654304-b37c230b-7c9d-4559-ad71-768a350c3d0e.jpg)

**Comment by deckerst** [comment]
I can't reproduce with a Samsung S10e, but I can with a Samsung S7. I'll take a closer look.

**Comment by xmha97** [comment]
> I can't reproduce

Use Microsoft Launcher

**Comment by xmha97** [comment]
> Do you know any open-source app that has a full frame widget with the MS launcher?

Yes, the Persian Calendar app:
https://github.com/persian-calendar/persian-calendar

**Comment by deckerst** [comment]
TLDR; the MS Launcher is broken, they should fix it.

Here's a comparison of the Samsung and MS launchers showing widgets from Google Photos (`G`), Aves (`A`) and Persian Calendar (`P`):
<img width="200" src="https://user-images.githubusercontent.com/13898333/192838274-10fd85cd-b0fd-4c95-b4b9-d8262c6e5135.png" /><img width="200" src="https://user-images.githubusercontent.com/13898333/192838410-17bcf6e6-d951-41b7-9f55-4ba9ace090c2.png" />

The Persian Calendar highlights something interesting. The widget itself, with its background color, is full frame, but the custom drawing is limited to the widget size reported by the MS Launcher, so its drawing is the same width as the images in the Google Photos or Aves widgets.
So background colors, text and other elements that do not depend on the widget size can go full frame, but elements that need to know the widget size (like custom drawing or cropped/scaled images) do not go full frame because the MS Launcher gives the wrong size.

**Comment by xmha97** [comment]
I tested your widget in [Smart Launcher](https://play.google.com/store/apps/details?id=ginlemon.flowerfree), there are margins around like [Microsoft Launcher](https://play.google.com/store/apps/details?id=com.microsoft.launcher)!
Please test the widget again.

![Screenshot_2022-11-05-02-32-04-110_ginlemon flowerfree](https://user-images.githubusercontent.com/23485114/200086963-e7ce23b7-f3f9-43db-aad8-c3e863c6419b.jpg)

![Screenshot_2022-11-05-02-41-48-774_ginlemon flowerfree](https://user-images.githubusercontent.com/23485114/200087703-4c8b08db-a7ad-46ad-af80-72a5348d86a3.jpg)

**Comment by xmha97** [comment]
![Screenshot_2022-11-05-02-46-00-174_ginlemon flowerfree](https://user-images.githubusercontent.com/23485114/200088007-1ab61445-e1de-41f7-b498-f99c8617bed0.jpg)

**Comment by deckerst** [comment]
Thanks for following up, but my guess is that this launcher also provides the wrong metrics for custom drawing. Please try another app that has a widget with custom drawing (not just a plain color background), like Google Photos, and I bet you'll see the same issue.

**Comment by xmha97** [comment]
Google Photos widget is bad on my device, low corners!

It's default Xiaomi launcher:

![Screenshot_2022-11-05-03-13-00-739_com miui home](https://user-images.githubusercontent.com/23485114/200089973-1af18d53-698f-4967-adb0-8ff662ec5a11.jpg)

**Comment by deckerst** [comment]
Tried Smart Launcher. Again, wrong metrics given by the launcher, as i suspected.
![Screenshot_20221105_075644_Smart Launcher](https://user-images.githubusercontent.com/13898333/200107099-bf03e414-ecf2-4066-bfaf-6a0adf298611.png)
